src/parser/creative_linear_parser.js

```diff
@@ -206,69 +192,151 @@ export function parseCreativeLinear(creativeElement, creativeAttributes) {
+  );
+  mediaFile.maxBitrate = parseInt(
+    mediaFileElement.getAttribute('maxBitrate') || 0
+  );
+  mediaFile.width = parseInt(mediaFileElement.getAttribute('width') || 0);
+  mediaFile.height = parseInt(mediaFileElement.getAttribute('height') || 0);
+
+  let scalable = mediaFileElement.getAttribute('scalable');
+  if (scalable && typeof scalable === 'string') {
+    scalable = scalable.toLowerCase();
+    if (scalable === 'true') {
+      mediaFile.scalable = true;
+    } else if (scalable === 'false') {
+      mediaFile.scalable = false;
+    }
+  }
 
-      parserUtils
-        .childrenByName(iconElement, 'StaticResource')
-        .forEach(staticElement => {
-          icon.type = staticElement.getAttribute('creativeType') || 0;
-          icon.staticResource = parserUtils.parseNodeText(staticElement);
-        });
+  let maintainAspectRatio = mediaFileElement.getAttribute(
+    'maintainAspectRatio'
+  );
+  if (maintainAspectRatio && typeof maintainAspectRatio === 'string') {
+    maintainAspectRatio = maintainAspectRatio.toLowerCase();
+    if (maintainAspectRatio === 'true') {
+      mediaFile.maintainAspectRatio = true;
+    } else if (maintainAspectRatio === 'false') {
+      mediaFile.maintainAspectRatio = false;
+    }
+  }
+  return mediaFile;
+}
 
-      const iconClicksElement = parserUtils.childByName(
-        iconElement,
-        'IconClicks'
-      );
-      if (iconClicksElement) {
-        icon.iconClickThroughURLTemplate = parserUtils.parseNodeText(
-          parserUtils.childByName(iconClicksElement, 'IconClickThrough')
-        );
-        parserUtils
-          .childrenByName(iconClicksElement, 'IconClickTracking')
-          .forEach(iconClickTrackingElement => {
-            icon.iconClickTrackingURLTemplates.push({
-              id: iconClickTrackingElement.getAttribute('id') || null,
-              url: parserUtils.parseNodeText(iconClickTrackingElement)
-            });
-          });
-      }
+/**
+ * Parses the InteractiveCreativeFile element from VAST MediaFiles node.
+ * @param  {Object} interactiveCreativeElement - The VAST InteractiveCreativeFile element.
+ * @return {Object} - Parsed interactiveCreativeFile object.
+ */
+function parseInteractiveCreativeFile(interactiveCreativeElement) {
+  const interactiveCreativeFile = createInteractiveCreativeFile();
+  interactiveCreativeFile.type = interactiveCreativeElement.getAttribute(
+    'type'
+  );
+  interactiveCreativeFile.apiFramework = interactiveCreativeElement.getAttribute(
+    'apiFramework'
+  );
+  let variableDuration = interactiveCreativeElement.getAttribute(
+    'variableDuration'
+  );
+  if (variableDuration && typeof variableDuration === 'string') {
+    variableDuration = variableDuration.toLowerCase();
+    if (variableDuration === 'true') {
+      interactiveCreativeFile.variableDuration = true;
+    } else if (variableDuration === 'false') {
+      interactiveCreativeFile.variableDuration = false;
```

Comment:
don't we have a util to correctly parse boolean variables? If not you could make one as it is used in multiple places now

Reply:
parseBoolean exists. And used